### PR TITLE
start-qemu.sh: allow to specify number of CPUs and sockets via cmdline

### PR DESCRIPTION
**Pull request by jurobystricky**
The number of CPUs and sockets is presently hardcoded as 1 each.
To change any of these values, user needs to edit the shell script.

This patch adds support to pass the desired number of cores and sockets via
command line options. The default values remain as before.

Number of CPUs must be positive, no upper limit checks. The assumption here
is that the user knows what he/she is doing.
Number of sockets is limited to reasonable values of 1 or 2.
Both limits can be easily modified if needed.
Note we use the upper case letter 'S' to specify the number of sockets,
as the lower case letter 's' is used to specify serial terminal.

Example: Create VM with 16 CPU cores and 2 sockets (8 CPUs per socket):
 ./start-qemu.sh -i td-guest-rhel8.5.qcow2 -k vmlinuz-rhel8.5 -c 16 -S 2

The CPU topology can be verified in VM via the command 'lscpu'.

Signed-off-by: Juro Bystricky <juro.bystricky@intel.com>

### CODE REVIEW

File: start-qemu.sh [hunk 1]
```diff
@@ -221,6 +235,9 @@ process_args() {
     QEMU_CMD+=$PARAM_MACHINE
     QEMU_CMD+=" -device virtio-net-pci,netdev=mynet0"
 
+    # Specify CPU topology: cores and sockets
+    QEMU_CMD+=" -smp ${CORES},sockets=${SOCKET} "
```
Comment (kenplusplus) [inline start-qemu.sh:239]: 1. I think the first parameter for -smp is not CORES but vCPUS according to https://wiki.gentoo.org/wiki/QEMU/Options, I think we can still use "-c" but just change to vCPU?
2. Do we have use case to specify SOCKET? for virtual 2 socket? virtual NUMA? If there is no specific use case, then we can remove it.

Comment (jurobystricky) [reply start-qemu.sh:239]: The variables CORES and SOCKET existed in the script before this patch. This patch only allows changing their values via command line so the script does not need to be manually edited each time you want to change the values of these variables. 
**" -smp $CORES,sockets=$SOCKET"** was there before. CORES means number of vCPUs, maybe the variable could be renamed.
I had to move those variables out of the original QEMU command line place to a different place as it is not possible to expand variables recursively. Do you want to remove the SOCKET completely and only allow setting the number of CPUs (via -c)?  I don't see any particularly good reason to have more than 1 socket but I assume whoever created this variable had a reason for it.

As for the vCPU, the document says:

-**smp <NUMBER> - Specify the number of cores the guest is permitted to use. The number can be higher than the available cores on the host system**

Comment (kenplusplus) [reply start-qemu.sh:239]: It is history issue. So if you agree, let's
- Change $CORES => $VCPUS or $CPUS
- Remove SOCKETS

Comment (jurobystricky) [reply start-qemu.sh:239]: Sounds good. I'll create a new patch.

**Review by kenplusplus — APPROVED** [review]
LGTM, thanks @jurobystricky

**Pull request merged**